File: android/app/src/main/java/com/transporthq/app/data/models/Analytics.kt
package com.transporthq.app.data.models

import com.google.gson.annotations.SerializedName

data class AnalyticsDashboard(
    val totalTrips: Int = 0,
    val completedTrips: Int = 0,
    val cancelledTrips: Int = 0,
    val activeTrips: Int = 0,
    val completionRate: Double = 0.0,
    val totalCost: Double = 0.0,
    val totalMiles: Double = 0.0,
    val averageTripDuration: Double = 0.0,
    val tripsByDay: List<TripsByDay> = emptyList(),
    val tripsByStatus: List<TripsByStatus> = emptyList()
)

data class TripsByDay(val date: String = "", val count: Int = 0)
data class TripsByStatus(val status: String = "", val count: Int = 0)

data class DriverPerformance(
    @SerializedName("_id") val id: String = "",
    val name: String = "",
    val totalTrips: Int = 0,
    val completedTrips: Int = 0,
    val cancelledTrips: Int = 0,
    val averageRating: Double = 0.0,
    val totalMiles: Double = 0.0,
    val onTimeRate: Double = 0.0
)

data class CostAnalytics(
    val totalCost: Double = 0.0,
    val averageCostPerTrip: Double = 0.0,
    val costByDay: List<CostByDay> = emptyList()
)

data class CostByDay(val date: String = "", val cost: Double = 0.0)

data class DailyReport(
    val date: String = "",
    val totalTrips: Int = 0,
    val completedTrips: Int = 0,
    val totalRideRequests: Int = 0,
    val activeDrivers: Int = 0,
    val totalMiles: Double = 0.0,
    val totalCost: Double = 0.0,
    val trips: List<Trip> = emptyList()
)

data class TimecardEntry(
    val driverId: String = "",
    val driverName: String = "",
    val totalHours: Double = 0.0,
    val trips: Int = 0,
    val date: String = ""
)

data class VehicleUtilization(
    val vehicleId: String = "",
    val vehicleName: String = "",
    val totalTrips: Int = 0,
    val totalMiles: Double = 0.0,
    val utilizationRate: Double = 0.0
)

data class OnTimeReport(
    val totalTrips: Int = 0,
    val onTimeTrips: Int = 0,
    val lateTrips: Int = 0,
    val onTimeRate: Double = 0.0
)

data class CostAnalysisReport(
    val totalCost: Double = 0.0,
    val averageCostPerTrip: Double = 0.0,
    val costByVehicle: List<CostByVehicle> = emptyList()
)

data class CostByVehicle(val vehicleName: String = "", val cost: Double = 0.0)

data class OvertimeEntry(
    val driverName: String = "",
    val regularHours: Double = 0.0,
    val overtimeHours: Double = 0.0,
    val date: String = ""
)

data class MaintenanceRecord(
    @SerializedName("_id") val id: String = "",
    val vehicle: VehicleRef? = null,
    val type: String = "",
    val title: String = "",
    val description: String = "",
    val status: String = "scheduled",
    val scheduledDate: String? = null,
    val completedDate: String? = null,
    val cost: Double = 0.0,
    val performedBy: String = "",
    val notes: String = "",
    val createdAt: String? = null
)

data class RatingEntry(
    @SerializedName("_id") val id: String = "",
    val rating: Int = 0,
    val comment: String = "",
    val type: String = "",
    val ratedBy: PassengerRef? = null,
    val createdAt: String? = null
)

data class RatingCheck(
    val hasRated: Boolean = false
)

data class AllocationEntry(
    val driver: DriverRef? = null,
    val passenger: PassengerRef? = null,
    val status: String = "",
    val allocatedAt: String? = null
)

data class AllocationDriver(
    val driver: DriverRef? = null,
    val isAllocated: Boolean = false
)

data class ActiveCar(
    @SerializedName("_id") val id: String = "",
    val name: String = "",
    val driverName: String = "",
    val status: String = "",
    val availableSeats: Int = 0,
    val currentLocation: Coordinates? = null
)

data class ShuttleRoute(
    @SerializedName("_id") val id: String = "",
    val name: String = "",
    val type: String = "crew",
    val stops: List<ShuttleStop> = emptyList(),
    val frequency: Int = 30,
    val startTime: String = "",
    val endTime: String = "",
    val capacity: Int = 20,
    val assignedVehicle: VehicleRef? = null,
    val assignedDriver: DriverRef? = null,
    val isActive: Boolean = true,
    val daysActive: List<String> = emptyList(),
    val notes: String = "",
    val createdAt: String? = null
)

data class ShuttleStop(
    val name: String = "",
    val address: String = "",
    val lat: Double = 0.0,
    val lng: Double = 0.0,
    val estimatedTime: String = "",
    val order: Int = 0
)

data class ShuttleRun(
    @SerializedName("_id") val id: String = "",
    val route: ShuttleRoute? = null,
    val driver: DriverRef? = null,
    val vehicle: VehicleRef? = null,
    val scheduledDeparture: Long = 0,
    val status: String = "scheduled",
    val occupancy: Int = 0,
    val capacity: Int = 20,
    val createdAt: String? = null
)

data class TripTemplate(
    @SerializedName("_id") val id: String = "",
    val name: String = "",
    val description: String = "",
    val pickupLocation: String = "",
    val dropoffLocation: String = "",
    val driverId: String? = null,
    val vehicleId: String? = null,
    val notes: String = "",
    val createdAt: String? = null
)

data class DayTemplate(
    @SerializedName("_id") val id: String = "",
    val name: String = "",
    val description: String = "",
    val sourceDate: Long = 0,
    val entries: List<DayTemplateEntry> = emptyList(),
    val usageCount: Int = 0,
    val createdAt: String? = null
)

data class DayTemplateEntry(
    val type: String = "",
    val title: String = "",
    val pickupLocation: String = "",
    val dropoffLocation: String = "",
    val pickupTime: String = "",
    val passengerCount: Int = 1
)

File: ios/TransportHQ/Models/Analytics.swift
import Foundation

struct AnalyticsDashboard: Codable {
    let totalTrips: Int?
    let completedTrips: Int?
    let cancelledTrips: Int?
    let activeTrips: Int?
    let completionRate: Double?
    let totalCost: Double?
    let totalMiles: Double?
    let averageTripDuration: Double?
    let tripsByDay: [TripsByDayEntry]?
    let tripsByStatus: [TripsByStatusEntry]?
}

struct TripsByDayEntry: Codable { let date: String?; let count: Int? }
struct TripsByStatusEntry: Codable { let status: String?; let count: Int? }

struct DriverPerformanceEntry: Codable, Identifiable {
    let id: String
    let name: String?
    let totalTrips: Int?
    let completedTrips: Int?
    let cancelledTrips: Int?
    let averageRating: Double?
    let totalMiles: Double?
    let onTimeRate: Double?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, totalTrips, completedTrips, cancelledTrips, averageRating, totalMiles, onTimeRate }
}

struct DailyReportData: Codable {
    let date: String?; let totalTrips: Int?; let completedTrips: Int?
    let totalRideRequests: Int?; let activeDrivers: Int?; let totalMiles: Double?; let totalCost: Double?
}

struct TimecardEntryData: Codable, Identifiable {
    var id: String { driverId ?? UUID().uuidString }
    let driverId: String?; let driverName: String?; let totalHours: Double?; let trips: Int?; let date: String?
}

struct VehicleUtilizationData: Codable, Identifiable {
    var id: String { vehicleId ?? UUID().uuidString }
    let vehicleId: String?; let vehicleName: String?; let totalTrips: Int?; let totalMiles: Double?; let utilizationRate: Double?
}

struct CostAnalysisData: Codable {
    let totalCost: Double?; let averageCostPerTrip: Double?
}

struct MaintenanceRecordData: Codable, Identifiable {
    let id: String
    let vehicle: MaintenanceVehicle?; let type: String?; let title: String?; let description: String?
    let status: String?; let scheduledDate: String?; let completedDate: String?; let cost: Double?
    let performedBy: String?; let notes: String?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case vehicle, type, title, description, status, scheduledDate, completedDate, cost, performedBy, notes, createdAt }
}

struct MaintenanceVehicle: Codable {
    let id: String?; let name: String?; let plateNumber: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, plateNumber }
}

struct RatingEntryData: Codable, Identifiable {
    let id: String; let rating: Int?; let comment: String?; let type: String?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case rating, comment, type, createdAt }
}

struct RatingCheckData: Codable { let hasRated: Bool? }

struct AllocationEntryData: Codable, Identifiable {
    var id: String { driver?.id ?? UUID().uuidString }
    let driver: AllocationDriverRef?; let passenger: AllocationPassengerRef?; let status: String?; let allocatedAt: String?
}

struct AllocationDriverRef: Codable {
    let id: String?; let name: String?; let phone: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, phone }
}

struct AllocationPassengerRef: Codable {
    let id: String?; let name: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name }
}

struct AllocationDriverData: Codable {
    let driver: AllocationDriverRef?; let isAllocated: Bool?
}

struct ActiveCarData: Codable, Identifiable {
    let id: String; let name: String?; let driverName: String?; let status: String?; let availableSeats: Int?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, driverName, status, availableSeats }
}

struct ShuttleRouteData: Codable, Identifiable {
    let id: String; let name: String?; let type: String?; let stops: [ShuttleStopData]?
    let frequency: Int?; let startTime: String?; let endTime: String?; let capacity: Int?
    let isActive: Bool?; let daysActive: [String]?; let notes: String?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, type, stops, frequency, startTime, endTime, capacity, isActive, daysActive, notes, createdAt }
}

struct ShuttleStopData: Codable { let name: String?; let address: String?; let estimatedTime: String?; let order: Int? }

struct ShuttleRunData: Codable, Identifiable {
    let id: String; let route: ShuttleRouteData?; let status: String?; let occupancy: Int?; let capacity: Int?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case route, status, occupancy, capacity, createdAt }
}

struct TripTemplateData: Codable, Identifiable {
    let id: String; let name: String?; let description: String?; let pickupLocation: String?; let dropoffLocation: String?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, description, pickupLocation, dropoffLocation, createdAt }
}

struct DayTemplateData: Codable, Identifiable {
    let id: String; let name: String?; let description: String?; let usageCount: Int?; let createdAt: String?
    enum CodingKeys: String, CodingKey { case id = "_id"; case name, description, usageCount, createdAt }
}
